packages/connect-web/src/grpc-web-transport.ts

```diff
@@ -279,7 +279,7 @@ export function createGrpcWebTransport(
             }
             trailerReceived = true;
             const trailer = trailerParse(data);
-            validateTrailer(trailer);
+            validateTrailer(trailer, undefined);
```

Comment:
Shouldn't we pass headers here as well?

Reply:
For streaming requests, the `onHeader` is triggered and we do send the headers there, should we also add them here?

Reply:
The [behavior as documented for connect-go](https://github.com/connectrpc/connect-go/blob/45625455d7dc24049956c74e632c7a7060656cef/error.go#L225) suggests we should add them here.

This may or may not cause problems when we collect information for the conformance runner, and merge headers twice. Not sure what the best way forward is - perhaps we can take a look at the go conformance impl?

The fact that `onHeader` only triggers for streaming in case of an error seems less than ideal.

Reply:
I think we should follow connect-go and always return headers in metadata. We can tweak the conformance implementation if needed.

Reply:
👍  I think this may need clarification in the conformance runner. Let's discuss elsewhere, this change is reasonable on it's own.